packages/ui5-tooling-transpile/lib/task.js

```diff
@@ -119,6 +119,24 @@ module.exports = async function ({ workspace /*, dependencies*/, taskUtil, optio
 					//traceResolution: true,
 				};
 
+				// update the sources map to declare the modules with the full module name
+				for await (const resourcePath of Object.keys(sourcesMap)) {
+					const source = sourcesMap[resourcePath];
+					let moduleName = /^\/resources\/(.*)\.ts$/.exec(resourcePath)?.[1];
+					if (moduleName?.endsWith(".gen.d")) {
```

Comment:
Hmmm.. so the heuristics is: any module named "*.gen.d" is assumed to be one generated by the ts-interface-generator and hence already containing a module declaration, which has to be modified, though? Should maybe be written down somewhere.

Reply:
Adding JSDoc